app/Http/Middleware/CheckPermissions.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace App\Http\Middleware;
+
+use Illuminate\Http\Request;
+use Closure;
+use App\Role;
+use App\User;
+
+class CheckPermissions
+{
+    const CODE401 = 401;
+
+    /**
+     * Handle an incoming request.
+     *
+     * @param  \Illuminate\Http\Request $request
+     * @param  \Closure $next
+     * @return mixed
+     */
+    public function handle($request, Closure $next, $role)
+    {
+        $userId = self::getUserIdFromToken($request);
+        if ($userId == 0) {
+            return self::getUnauthorized();
+        } else {
+            $userPermissions = User::find($userId)->roles;
+            $roleArray = explode(";", $role);
```

Comment:
$role can contains more roles, then it should have a different name because it is confusing